### ISSUE TITLE
[Sandbox] Kubemarine

### ISSUE THREAD

**Issue by koryaga** [cncf/sandbox#16]
### Application contact emails

Igor Kokorin <Igor.Kokorin@NetCracker.com>, Sergey Kryazhev <sergey.kryazhev@netcracker.com>

### Project Summary

Kubemarine is Kubernetes cluster deployment and maintenance tool

### Project Description

Kubemarine is an open source, lightweight and powerful management tool built for end-to-end Kubernetes cluster deployment and maintenance. It is applicable for many purposes like simple and quick onboarding Kubernetes on local and production environments in different HA schemes depending on your aims, budget, and capabilities. Together with simplicity, Kubemarine can be a very flexible and customizable tool covering specific configurability cases on both deployment and maintenance stages. This library provides powerful CLI commands, as well as can be customized using a Python extension API.

### Org repo URL

https://github.com/Netcracker

### Project repo URL

https://github.com/Netcracker/KubeMarine

### Roadmap

see below

### Roadmap context

- Geo Redundancy cluster deployment support. Several Kubernetes clusters connected together based on Calico-BGP/Submariner.
- Graphical User Interface
- _check_ procedures extension. Add more "checks", including support of Public cloud.
- python modularization. Make Kubemarine as a python module to be managed by pip.
- Provide more API for Kubemarine customization

### Contributing Guide

https://github.com/Netcracker/KubeMarine/blob/main/CONTRIBUTING.md

### Code of Conduct (CoC)

https://github.com/Netcracker/KubeMarine/blob/main/CODE-OF-CONDUCT.md

### Maintainers file

https://github.com/Netcracker/KubeMarine/blob/main/MAINTAINERS.md

### IP Policy

- [X] If the project is accepted, I agree the project will follow the CNCF IP Policy

### Trademark and accounts

- [X] If the project is accepted, I agree to donate all project trademarks and accounts to the CNCF

### Why CNCF?

We use Kubemarine in different on premise Kubernetes installations and it is a simple (single inventory file) , straightforward and robust way to deploy and maintain Kubernetes clusters.
Being a part of CNCF demonstrates maturity of Kubemarine, with its practical expirience and strong community which stands behind it.
It will promote other engeneers to use and contribute Kubemarine distribution.

### Benefit to the Landscape

Kubemarine provides good combinations of  easy usage and enterprise maturity. In combination with pure Python support this may look like a simple, but robust Kubernetes distribution ready  for simple quick start and future enterprise growth valuable for other CNCF members. 

### Similar projects

In some terms kublr and rancher

### Product or Service to Project separation

N/A

### Project presentations

https://dev.to/netcracker/deploy-and-maintain-kubernetes-with-kubemarine-functional-overview-1mkm

**Comment by dims** [comment]
@koryaga feels like you are a bit early in your journey as a kubernetes installer. can you please talk a bit about how you ended up writing a new one and how your approach is different from the other installers?

**Comment by koryaga** [comment]
Hi @dims.
The focus was on simplicity. We need an open , highly customisable , extendable , Python based solution. This is important part of our internal needs. That's why we are here ) 
Keeping in mind that we are going soon to make kubemarine as a pure Python module , this would nicely suits into Python infrastructure to rollout and manage Kubernetes distribution right from the Python code.
Another point is  to use latest,  tested , OOB 3rd party pack (k8s, nginx as ingress-controller, calico as CNI, and so on) and rapidly update , validate and deliver their  latest version , since we have easy 3rd party update lifecycle.
In addition kubemarine provides: multiple deployment schemas (AllInOne, MiniHA, FullHA), plugin support: shell, helm (Helm charts might be installed automatically), seamless k8s upgrade procedure, preinstall  and postinstall checks and growing number of  maintenance procedures.

**Comment by koryaga** [comment]
Hi, just would like to mention we have released Python module distribution.  https://pypi.org/project/kubemarine/   Now it is possible to use Kubemarine in a more native way in Python infrastructure.

**Comment by raravena80** [comment]
FYI. The maintainers are requesting more feedback as to why the project was voted down for sandbox. Thx.

cc: @koryaga

**Comment by TheFoxAtWork** [comment]
Thanks @raravena80 !

We talked through this on the recording but I am happy to summarize here:
* at the [beginning of the recording](https://youtu.be/qt0cfsuGfVo?t=40) we discussed and relayed the intent in sandbox, what we look for, and what it does.  We consider its viability on becoming fully mature with widespread adoption like other graduated projects, the need to experiment within the space, need for vendor neutrality, the value and expectations for becoming a CNCF project, and existing gaps in the ecosystem. Overall this discussion is around 9 mins long.
* We began [discussing Kubemarine about 25 minutes in](https://youtu.be/qt0cfsuGfVo?t=1531) - discussion focused on scope of the project being narrow which means the likelihood of prolific adoption is lower. Through the discussion we felt Kubemarine was positioned as a solution-oriented project and not in need of experimentation. The project may benefit from discussion/presentation at a Kubernetes SIG to increase awareness of its capabilities to adopters looking for something in that particular space. Therefore we declined to accept it.
